hail/python/test/hail/conftest.py

```diff
@@ -65,19 +78,15 @@ def pytest_runtest_makereport(item, call):
     return report
 
 
-@pytest.fixture(autouse=True)
-def reinitialize_hail_for_each_qob_test(init_hail, request):
-    if isinstance(current_backend(), ServiceBackend):
-        hl_stop_for_test()
-        hl_init_for_test(app_name=request.node.name)
-        new_backend = current_backend()
-        assert isinstance(new_backend, ServiceBackend)
-        yield
-        if new_backend._batch_was_submitted:
-            batch = new_backend._batch
-            report: Dict[str, CollectReport] = request.node.stash[test_results_key]
-            if any(r.failed for r in report.values()):
-                log.info(f'cancelling failed test batch {batch.id}')
-                hail_event_loop().run_until_complete(batch.cancel())
-    else:
-        yield
+@pytest.fixture
+def reinitialize_hail_for_testing(init_hail, request):
+    hl_stop_for_test()
+    hl_init_for_test(app_name=request.node.name)
+    yield
+    new_backend = current_backend()
+    if isinstance(new_backend, ServiceBackend) and new_backend._batch_was_submitted:
```

Comment:
IIUC this fixture will now only be added when `choose_backend() == 'batch'`, so can we remove this isinstance check?

Reply:
oh I changed this to be a generic fixture so anyone can call this.